spring-data-azure-documentdb/src/main/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplate.java

```diff
@@ -0,0 +1,221 @@
+
+            documentDbFactory.getDocumentClient().createDocument(collection.getSelfLink(),
+                    document, null, false);
+            return objectToSave;
+        } catch (DocumentClientException e) {
+            e.printStackTrace();
+            throw new RuntimeException(e.getMessage());
+        }
+    }
+
+    public <T> T findById(Object id, Class<T> entityClass) {
+        final String query = "SELECT * from c where c.id='" + id.toString() + "'";
+
+        final List<Document> results = documentDbFactory.getDocumentClient()
+                .queryDocuments("dbs/" + databaseName + "/colls/" + getCollectionName(entityClass), query, null)
+                .getQueryIterable().toList();
+
+        if (results == null || results.size() == 0) {
+            return null;
+        } else {
+
+            final Document d = results.get(0);
+            return dbConverter.convertFromDocument(d, entityClass);
+        }
+
+    }
+
+    public <T> void delete(T objectToRemove) {
+        throw new UnsupportedOperationException("not supported");
+    }
+
+    private Database getDb(String dbName) {
+        final String query = "SELECT * from root r where r.id='" + dbName + "'";
+
+        try {
+            final List<Database> dbList = documentDbFactory.getDocumentClient()
+                    .queryDatabases(query, null).getQueryIterable().toList();
+
+            if (dbList.size() > 0) {
+                return dbList.get(0);
+            } else {
+                // create new database
+                Database db = new Database();
+                db.setId(dbName);
+
+                final Resource resource = documentDbFactory.getDocumentClient()
+                        .createDatabase(db, null).getResource();
+
+                if (resource instanceof Database) {
+                    db = (Database) resource;
+                }
+                return db;
+            }
+        } catch (DocumentClientException ex) {
+            throw new RuntimeException(ex.getMessage());
+        }
+    }
+
+    public DocumentCollection createCollection(String collectionName, RequestOptions collectionOptions) {
+        DocumentCollection collection = new DocumentCollection();
+        collection.setId(collectionName);
+
+        try {
+
+            final Resource resource = documentDbFactory.getDocumentClient()
+                    .createCollection("dbs/" + databaseName, collection, collectionOptions)
+                    .getResource();
+            if (resource instanceof DocumentCollection) {
+                collection = (DocumentCollection) resource;
+            }
+            return collection;
+        } catch (DocumentClientException e) {
+            throw new RuntimeException(e.getMessage());
+        }
+
+    }
+
+    private DocumentCollection getCollection(String dbName, String collectionName) {
+        final String query = "SELECT * FROM root r where r.id='" + collectionName + "'";
```

Comment:
From performance standpoint its better to use Stringbuilder. if we call getCollection 1000 times then 1000 objects will be stored in string pool and wont be available for garbage collection. Instead if we use StringBuilder it will be available for garbage collection